lib/controller.js

```diff
@@ -45,152 +33,134 @@ const getEmoji = (repoLength) => {
     return '';
 };
 
-// handle search and display statistics
-exports.index = (req, res) => {
-    const username = req.query.username;
-    const headers = {
-        'User-Agent': `${username}`
-    };
-
-    const userOptions = {
-        url: `https://api.github.com/users/${username}?client_id=${client_id}&client_secret=${client_secret}`,
-        headers
-    };
-
-    const repoOptions = {
-        url: `https://api.github.com/users/${username}/repos?client_id=${client_id}&client_secret=${client_secret}&per_page=100`,
-        headers
-    };
+const reqUser = (username) => {
+    const url = `https://api.github.com/users/${username}?client_id=${client_id}&client_secret=${client_secret}`;
+    const headers = { 'User-Agent': `${username}` };
+    return axios.get(url, { headers });
+};
 
-    let languages = {},
-        repos = '',
-        msg = '',
-        avatar,
-        statement;
+const reqRepos = (username, numberOfPages) => {
+    const headers = { 'User-Agent': `${username}` };
+    const url = (page) =>
+        `https://api.github.com/users/${username}/repos?client_id=${client_id}&client_secret=${client_secret}&per_page=100&page=${page}`;
+    const requests = [];
+    for (let i = 1; i <= numberOfPages; i++) {
+        requests.push(axios.get(url(i), { headers }));
+    }
+    return requests;
+};
 
+exports.index = (req, res) => {
+    const { username } = req.query;
+    let avatar, msg, repos, statement;
```

Comment:
I thought would be better use let once like : `let avatar, msg, repos, statement, languages = {};

Reply:
true, good catch